txrelayer/txrelayer.go

```diff
@@ -0,0 +1,189 @@
+package txrelayer
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/umbracle/ethgo"
+	"github.com/umbracle/ethgo/jsonrpc"
+	"github.com/umbracle/ethgo/wallet"
+)
+
+const (
+	defaultGasPrice = 1879048192 // 0x70000000
+	defaultGasLimit = 5242880    // 0x500000
+)
+
+var (
+	errNoAccounts               = errors.New("no accounts registered")
+	errNonSignedTxnNotSupported = errors.New("sending non signed transactions supported only for local account mode")
+)
+
+type TxRelayer interface {
```

Comment:
I would prefer if we move the interfaces to the place where we need them. In the end, an interface defines a level of abstraction, it should be the module that uses the relayer the one that describes which is the abstraction it requires.

This is, if the `consensusRuntime` needs the relayer, it should be the one defining the interface of that relayer, not relying on this.

Reply:
But in this case it would mean duplicate interface on two places:
1. `polybft` package (`checkpointManager` uses it) and
2. `initcontracts` package (`rootchain init-contracts` uses it).

I don't like duplicating code (e.g. defining same interface twice), also it is good practice to hide implementation details from the caller (that's why I have introduced interface in the first place) therefore I would prefer to keep it in `txrelayer` package.